packages/admin/src/middleware.page.ts

```diff
@@ -11,6 +11,16 @@ const authenticateRequest = async (req: NextRequest, res: NextResponse) => {
   const authCookie = req.cookies.get('session_id');
   const userJwtCookie = req.cookies.get(process.env.JWT_COOKIE_NAME);
 
+  await csrfMiddleware(req, res);
+  //means that the user is logged in but not as an admin/superAdmin(otherwise they would have had the session_id cookie set)
+  if (!authCookie && userJwtCookie) {
+    const url = getLoginUrl({ redirectTo404: true });
+    logger.info(
+      `User is not an admin - redirecting it to appropriate app 404 page`
+    );
+    return NextResponse.redirect(url);
```

Comment:
```suggestion
    return NextResponse.redirect(url, { status: 302 });
```
probably best to follow this pattern as with the others